e2e-tests/src/driver.rs

```diff
@@ -95,56 +103,54 @@ impl Driver {
+                    ProtocolSandboxConfig::Stellar(_) => Protocol::Stellar,
+                    ProtocolSandboxConfig::Near(_) => Protocol::Near,
+                    ProtocolSandboxConfig::Icp(_) => Protocol::Icp,
+                    ProtocolSandboxConfig::Ethereum(_) => Protocol::Ethereum,
                 };
 
-                if config_protocol_name == protocol_name {
-                    let sandbox_env = match sandbox_cfg {
-                        ProtocolSandboxConfig::Stellar(config) => {
-                            ProtocolSandboxEnvironment::Stellar(StellarSandboxEnvironment::init(
-                                config.clone(),
-                            )?)
-                        }
-                        ProtocolSandboxConfig::Near(config) => ProtocolSandboxEnvironment::Near(
-                            NearSandboxEnvironment::init(config.clone()).await?,
-                        ),
-                        ProtocolSandboxConfig::Icp(config) => ProtocolSandboxEnvironment::Icp(
-                            IcpSandboxEnvironment::init(config.clone())?,
-                        ),
-                        ProtocolSandboxConfig::Ethereum(config) => {
-                            ProtocolSandboxEnvironment::Ethereum(EthereumSandboxEnvironment::init(
-                                config.clone(),
-                            )?)
-                        }
-                    };
-
-                    initialized_protocols.insert(protocol_name.clone(), sandbox_env);
-                    break;
+                if &config_protocol != protocol {
+                    continue;
                 }
+
+                let sandbox_env = match sandbox_cfg {
+                    ProtocolSandboxConfig::Stellar(config) => ProtocolSandboxEnvironment::Stellar(
+                        StellarSandboxEnvironment::init(config.clone())?,
+                    ),
+                    ProtocolSandboxConfig::Near(config) => ProtocolSandboxEnvironment::Near(
+                        NearSandboxEnvironment::init(config.clone()).await?,
+                    ),
+                    ProtocolSandboxConfig::Icp(config) => ProtocolSandboxEnvironment::Icp(
+                        IcpSandboxEnvironment::init(config.clone())?,
+                    ),
+                    ProtocolSandboxConfig::Ethereum(config) => {
+                        ProtocolSandboxEnvironment::Ethereum(EthereumSandboxEnvironment::init(
+                            config.clone(),
+                        )?)
+                    }
+                };
+
+                initialized_protocols.insert(*protocol, sandbox_env);
+                break;
             }
         }
 
         for protocol_name in &self.environment.protocols {
-            let protocol_name = protocol_name.to_string().to_lowercase();
-            let protocol_path = protocols_dir.join(&protocol_name);
+            let protocol_path = protocols_dir.join(&protocol_name.to_string().to_lowercase());
```

Comment:
then this can become:

```suggestion
            let protocol_path = protocols_dir.join(protocol_name.as_str());
```